dataservice/api/diagnosis/resources.py

```diff
@@ -0,0 +1,117 @@
+from flask import (
+    abort,
+    request
+)
+from flask.views import MethodView
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound
+from marshmallow import ValidationError
+
+from dataservice.extensions import db
+from dataservice.api.errors import handle_integrity_error
+from dataservice.api.diagnosis.models import Diagnosis
+from dataservice.api.diagnosis.schemas import DiagnosisSchema
+
+
+class DiagnosisAPI(MethodView):
+    """
+    Diagnosis REST API
+    """
+
+    def get(self, kf_id):
+        """
+        Get a diagnosis by id or get all diagnoses
+
+        Get a diagnosis by Kids First id or get all diagnoses if
+        Kids First id is None
+        """
+        # Get all
+        if kf_id is None:
+            d = Diagnosis.query.all()
+            return DiagnosisSchema(many=True).jsonify(d)
+        # Get one
+        else:
+            try:
+                d = Diagnosis.query.filter_by(kf_id=kf_id).one()
+            # Not found in database
+            except NoResultFound:
+                abort(404, 'could not find {} `{}`'
+                      .format('diagnosis', kf_id))
+            return DiagnosisSchema().jsonify(d)
+
+    def post(self):
+        """
+        Create a new diagnosis
+        """
+        body = request.json
+        try:
```

Comment:
This should be two `try/catch` blocks, one for the validation, one for the transaction.